db/migrate/20151124211951_create_incarnator_accounts_roles.rb

```diff
@@ -0,0 +1,19 @@
+class CreateIncarnatorAccountsRoles < ActiveRecord::Migration
+  def change
+    create_table :incarnator_accounts_roles do |t|
+      t.references :account, null: false
+      t.references :role,    null: false
+
+      t.timestamps null: false
+    end
+
+    add_index :incarnator_accounts_roles, :account_id
```

Comment:
don't see sense for separated indexes on both account_id and user_id, the unique index is sufficient as for me